src/OpenDiffix.Core/Analyzer.fs

```diff
@@ -309,73 +309,63 @@ let private bucketExpand aidColumnsExpression =
 
-  let nonConstantExpressions =
-    selectedExpressions
-    |> List.filter (
-      function
-      | Constant _ -> false
-      | _ -> true
-    )
+  let noGrouping = List.isEmpty selectQuery.GroupBy
+  let noAggregation = selectedExpressions |> List.forall Expression.isScalar
+
+  let selectQuery =
+    if noGrouping && noAggregation then
+      // Non-aggregating query; group implicitly and expand.
+      { selectQuery with
+          TargetList =
+            {
+              Expression = bucketExpand aidColumnsExpression
+              Alias = ""
+              Tag = JunkTargetEntry
+            }
+            :: selectQuery.TargetList
+          GroupBy = selectedExpressions
+      }
+    else
+      selectQuery
 
-  let doesGrouping = List.isEmpty selectQuery.GroupBy |> not
-  let doesAggregation = selectedExpressions |> List.forall Expression.isScalar |> not
-  let onlyConstantsSelected = List.isEmpty nonConstantExpressions
-
-  if not doesGrouping && not doesAggregation then
-    // Non-grouping, non-aggregate query; group implicitly and expand
-    let having =
-      if not onlyConstantsSelected then
-        Expression.makeAnd lowCountFilter selectQuery.Having
-      else
-        // All selected expressions are constants; no low-count filter in line with global anonymized count.
-        selectQuery.Having
-
-    { selectQuery with
-        TargetList =
-          {
-            Expression = bucketExpand aidColumnsExpression
-            Alias = ""
-            Tag = JunkTargetEntry
-          }
-          :: selectQuery.TargetList
-        // No need to group implicitly by constants, which are invalid bucket definitions anyway.
-        GroupBy = nonConstantExpressions
-        Having = having
-    }
-  else if not doesGrouping && doesAggregation then
-    // Non-grouping aggregate query; do nothing.
-    selectQuery
-  else
-    // Grouping query; add LCF to HAVING
-    { selectQuery with
-        Having = Expression.makeAnd lowCountFilter selectQuery.Having
-    }
+  // Ignore constant expressions during grouped aggregation.
+  let groupBy = List.filter (isConstant >> not) selectQuery.GroupBy
 
-let private rewriteQuery anonParams (selectQuery: SelectQuery) =
+  let having =
+    if List.isEmpty groupBy then
+      // No need to apply the LCF to the global bucket.
+      selectQuery.Having
+    else
+      // Apply the LCF to the output buckets.
+      [ aidColumnsExpression ]
+      |> Expression.makeAggregate DiffixLowCount
+      |> Expression.makeNot
+      |> Expression.makeAnd selectQuery.Having
+
+  { selectQuery with GroupBy = groupBy; Having = having }
+
+let private compileQuery anonParams (selectQuery: SelectQuery) =
   let rangeColumns = collectRangeColumns anonParams selectQuery.From
   let aidColumnsExpression = makeAidColumnsExpression rangeColumns
 
-  let isAnonymizing =
-    anonParams.AccessLevel <> Direct
-    && aidColumnsExpression |> Expression.unwrapListExpr |> List.isEmpty |> not
+  let directAccess =
```

Comment:
I'm not necessarily a fan of inverting this condition and calling it `directAccess`. That can be true even despite `AccessLevel <> Direct`, so despite being equivalent, might give rise to some confusion.

Also note, that you're still using the `anonymizing` word in `validate/compileAnonymizingQuery`.

Reply:
Inverting the condition name simplified a bit the corresponding checks. 
I don't feel strongly about it so I will leave it up to you to revert the change or maybe use a different name like `isStandardQuery`.